swatch-contracts/src/main/java/com/redhat/swatch/contract/security/RhIdentityHeaderAuthenticationMechanism.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package com.redhat.swatch.contract.security;
+
+import io.quarkus.security.AuthenticationFailedException;
+import io.quarkus.security.identity.IdentityProviderManager;
+import io.quarkus.security.identity.SecurityIdentity;
+import io.quarkus.security.identity.request.AuthenticationRequest;
+import io.quarkus.vertx.http.runtime.security.ChallengeData;
+import io.quarkus.vertx.http.runtime.security.HttpAuthenticationMechanism;
+import io.smallrye.mutiny.Uni;
+import io.vertx.ext.web.RoutingContext;
+import java.util.Set;
+import javax.annotation.Priority;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Alternative;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Allows authentication attempts via x-rh-identity header.
+ *
+ * <p>Handles the decoding of the header; then delegates validation of the properly formed identity
+ * to {@link RhIdentityIdentityProvider}.
+ *
+ * @see RhIdentityIdentityProvider
+ * @see RhIdentityAuthenticationRequest
+ */
+@Slf4j
+@Alternative
+@Priority(1)
+@ApplicationScoped
+public class RhIdentityHeaderAuthenticationMechanism implements HttpAuthenticationMechanism {
+  private static final String RH_IDENTITY_HEADER = "x-rh-identity";
+
+  @Override
+  public Uni<SecurityIdentity> authenticate(
+      RoutingContext context, IdentityProviderManager identityProviderManager) {
+    String xRhIdentityHeader = context.request().headers().get(RH_IDENTITY_HEADER);
+    if (xRhIdentityHeader == null) {
+      // no authentication attempted
+      return Uni.createFrom().nullItem();
```

Comment:
`Uni.createFrom().nullItem()` and `Uni.createFrom().optional(Optional.empty())` are equivalent, because the both emit `null` (see  https://javadoc.io/static/io.smallrye.reactive/mutiny/0.11.0/index.html?io/smallrye/mutiny/Uni.html)